File: frontend/src/types/medicine.ts
export interface MedicineListItem {
  id: number
  name: string
  genericName: string
  dosageForm: string
  specification: string
  baseUnit: string
  packageUnit: string
  manufacturer: string
  purchasePrice: number
  stockThreshold: number
  stockThresholdInPackages: number
  status: 'ACTIVE' | 'INACTIVE'
  barcodes: string[]
}

export interface MedicineDetail {
  id: number
  name: string
  genericName: string
  dosageForm: string
  specification: string
  baseUnit: string
  packageUnit: string
  manufacturer: string
  purchasePrice: number
  stockThreshold: number
  stockThresholdInPackages: number
  status: 'ACTIVE' | 'INACTIVE'
  pinyinAbbr: string
  remark: string
  createdAt: string
  updatedAt: string
  conversions: ConversionItem[]
  barcodes: BarcodeItem[]
}

export interface ConversionItem {
  id: number
  fromUnit: string
  toUnit: string
  factor: number
}

export interface BarcodeItem {
  id: number
  barcode: string
  remark: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  page: number
  size: number
}

export interface SaveMedicinePayload {
  name: string
  genericName?: string
  dosageForm?: string
  specification?: string
  baseUnit: string
  packageUnit?: string
  manufacturer?: string
  purchasePrice: number
  stockThreshold?: number | null
  remark?: string
}

export interface SaveConversionPayload {
  fromUnit: string
  toUnit: string
  factor: number
}

export interface SaveBarcodePayload {
  barcode: string
  remark?: string
}

File: backend/src/main/java/com/fafeng/clinic/medicine/entity/Medicine.java

package com.fafeng.clinic.medicine.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

import lombok.Data;
import lombok.NoArgsConstructor;

@TableName("medicine")
@Data
@NoArgsConstructor
public class Medicine {

    public static final String STATUS_ACTIVE = "ACTIVE";
    public static final String STATUS_INACTIVE = "INACTIVE";
    public static final String STATUS_DELETED = "DELETED";

    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String genericName;
    private String dosageForm;
    private String specification;
    private String baseUnit;
    private String packageUnit;
    private String manufacturer;
    private BigDecimal purchasePrice;
    private BigDecimal suggestedRetailPrice;
    private BigDecimal stockThreshold;
    private String pinyinAbbr;
    private String remark;
    private String status;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;

}
